File: ShoppingPlatform-SPA/src/app/_models/product.ts
import { DecimalPipe } from '@angular/common';
import { Category } from './category';
import { User } from './user';

export interface Product{
    id: number;
    title: string;
    description: string;
    categoryName: string;
    price: number;
    seller: User;
    addedDate: Date;
}

File: ShoppingPlatform.API/Entities/Product.cs
using System;

namespace ShoppingPlatform.API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public Category Category { get; set; }
        public decimal Price { get; set; }
        public User Seller { get; set; }
        public DateTime AddedDate { get; set; }

        public Product()
        {
            AddedDate = DateTime.Now;
        }
    }
}